test/spec/modules/staqAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,350 @@
+      events.getEvents.restore();
+    });
+
+    it('should be configurable', function () {
+      adapterManager.registerAnalyticsAdapter({
+        code: 'staq',
+        adapter: analyticsAdapter
+      });
+
+      adapterManager.enableAnalytics({
+        provider: 'staq',
+        options: {
+          connId: 777,
+          queueTimeout: 1000,
+          url: 'http://localhost/prebid'
+        }
+      });
+
+      // expect(analyticsAdapter.context).to.have.property('connId', 'tag.staqadapter.com');
+      expect(analyticsAdapter.context).to.have.property('connectionId', 777);
+    });
+
+    it('should handle auction init event', function () {
+      events.emit(CONSTANTS.EVENTS.AUCTION_INIT, {config: {}, timeout: 3000});
+      const ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(1);
+      expect(ev[0]).to.be.eql({event: 'auctionInit', auctionId: undefined});
+    });
+
+    it('should handle bid request event', function () {
+      events.emit(CONSTANTS.EVENTS.BID_REQUESTED, REQUEST);
+      const ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(2);
+      expect(ev[1]).to.be.eql({
+        adUnitCode: 'container-1',
+        auctionId: '5018eb39-f900-4370-b71e-3bb5b48d324f',
+        event: 'bidRequested',
+        adapter: 'AppNexus',
+        bidderName: 'AppNexus'
+      });
+    });
+
+    it('should handle bid response event', function () {
+      events.emit(CONSTANTS.EVENTS.BID_RESPONSE, RESPONSE);
+      const ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(3);
+      expect(ev[2]).to.be.eql({
+        adId: '208750227436c1',
+        event: 'bidResponse',
+        adapter: 'AppNexus',
+        bidderName: 'AppNexus',
+        auctionId: '5018eb39-f900-4370-b71e-3bb5b48d324f',
+        adUnitCode: 'container-1',
+        cpm: 0.015,
+        timeToRespond: 0.443,
+        height: 250,
+        width: 300,
+        bidWon: false,
+      });
+    });
+
+    it('should handle timeouts properly', function() {
+      events.emit(CONSTANTS.EVENTS.BID_TIMEOUT, bidTimeoutArgsV1);
+
+      const ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(5); // remember, we added 2 timeout events
+      expect(ev[3]).to.be.eql({
+        adapter: 'AppNexus',
+        auctionId: '66529d4c-8998-47c2-ab3e-5b953490b98f',
+        bidderName: 'AppNexus',
+        event: 'adapterTimedOut'
+      })
+    });
+
+    it('should handle winning bid', function () {
+      events.emit(CONSTANTS.EVENTS.BID_WON, RESPONSE);
+      const ev = analyticsAdapter.context.queue.peekAll();
+      expect(ev).to.have.length(6);
+      // let secondCallArgs = ajaxStub.secondCall.args[0];
+      // let ev = JSON.parse(secondCallArgs);
+      // console.log('AUCTION WON EVENT SHAPE ' + JSON.stringify(ev));
+      // let firstEv = JSON.parse(ajaxStub.firstCall.args)
+      // console.log('FIRST CALL ARGS ' + JSON.stringify(firstEv));
```

Comment:
Can you remove this commented code if it's no longer needed?

Reply:
@jsnellbaker thanks for your review and your help.  I think I've got everything you asked for cleared up.